code/game/atoms.dm

```diff
@@ -428,7 +428,28 @@ its easier to just keep the beam vertical.
 			to_chat(user, harm_label_examine[1])
 		else
 			to_chat(user, harm_label_examine[2])
-	return
+
+	var/obj/item/device/camera_bug/bug = locate() in src
+	if(bug)
+		var/this_turf = get_turf(src)
+		var/user_turf = get_turf(user)
+		var/distance = get_dist(this_turf, user_turf)
+		if(Adjacent(user))
+			to_chat(user, "<a href='?src=\ref[src];bug=\ref[bug]'>There's something hidden in there.</a>")
+		else if(isobserver(user) || prob(100 / (distance + 2)))
```

Comment:
If, when `mob` is adjacent, distance becomes 1, couldn't you remove the `+2` put this in the main if clause ?
If distance becomes 0, you could do the same but replace `+2` with `+1`.